lib/msf/core/post/file.rb

```diff
@@ -500,20 +502,45 @@ def rm_rf(*remote_dirs)
   alias :file_rm :rm_f
   alias :dir_rm :rm_rf
 
+  def get_separator(file_path)
+    if file_path.include?('\\')
+      '\\'
+    else
+      '/'
+  end
```

Comment:
This doesn't look like it's being used anywhere so it should be safe to remove. If this functionality were going to be included anyways, it should be relying on the platform instead of guessing based on inspecting the string.

Reply:
Agree on removal if no longer used however this looks to be just have been accepted of a suggestion snip that still needs an update on line 528 to call the new function name.

On detection based on platform I am not convinced, in the original usage for this is to add a separator to an existing input path from the caller.  Reaching into the session for platform assumes the caller is always programmatically generating the string and most shells will support both formats as mentioned in the [original suggestion](https://github.com/rapid7/metasploit-framework/pull/15332#discussion_r664570238).

Reply:
In the last commit co-authored by @jmartin-r7 , the name of the method got changed. This method is used to get the separator so that we can extract the file_name from the path and then concate if new_file is a directory.

Reply:
Resolving this because I am not using this method anymore. :+1: